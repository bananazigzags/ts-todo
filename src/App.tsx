import './App.css';
import { Dispatch, useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { Navbar } from './components/Navbar/Navbar';
import { TodoInput } from './components/TodoInput/TodoInput';
import { TodoList } from './components/TodoList/TodoList';
import { fetchTodos } from './redux/todos/asyncActions';

export const App: React.FC = () => {
  const dispatch: Dispatch<any> = useDispatch()
  useEffect(() => {
    dispatch(fetchTodos())
  }, [])

  return (
    <div className="App">
      <Navbar />
      <TodoInput />
      <TodoList />
    </div>
  );
}

export default App;
