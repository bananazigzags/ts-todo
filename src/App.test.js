import React from 'react';
import { screen, render, getByRole } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from './redux/store/store';
import App from './App';

describe("app component", () => {

  it('renders app', () => {
    const { container } = render(
      <Provider store={store}>
        <App />
      </Provider>
    ); 
    expect(container).toMatchSnapshot();
  });

  it('renders app name', () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    const heading = screen.getByText(/дела/i); 
    expect(heading).toBeInTheDocument();
  });

  it('renders three filters', () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    screen.debug()
    const filters = screen.getAllByRole("listitem"); 
    expect(filters.length).toBe(3);
  });

  it('renders input box', () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    const inputBox = screen.getByRole("textbox"); 
    expect(inputBox).toBeInTheDocument();
  });

})