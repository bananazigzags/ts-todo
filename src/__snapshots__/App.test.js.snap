// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component renders app 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="navbar"
    >
      <h1>
        ДЕЛА
      </h1>
      <ul
        class="filters"
      >
        <li
          id="pending"
        >
          В работе
        </li>
        <li
          id="favorites"
        >
          Избранные
        </li>
        <li
          id="completed"
        >
          Выполненные
        </li>
      </ul>
    </div>
    <div
      class="todo-input-block"
    >
      <div
        class="todo-input"
      >
        <input
          type="text"
          value=""
        />
        <button
          class="add-btn"
          type="button"
        >
           Добавить
        </button>
      </div>
    </div>
    <div
      class="todos"
    />
  </div>
</div>
`;
